src/authhelper.cpp

```diff
@@ -0,0 +1,36 @@
+#include <QObject>
+#include <QProcess>
+
+#include <KAuth>
+#include <QDebug>
+#include <QEventLoop>
+#include <QTimer>
+
+class AuthHelper : public QObject
+{
+    Q_OBJECT
+public:
+    AuthHelper() = default;
+
+public slots:
+    KAuth::ActionReply elevate(const QVariantMap& args)
+    {
+        QEventLoop loop;
+        KAuth::ActionReply reply = KAuth::ActionReply::SuccessReply();
```

Comment:
use `return KAuth::ActionReply::SuccessReply` below directly, but also handle errors properly by returning non-success replies as needed

Reply:
the only way this could fail during my testing was when the elevate script was not available. I added a check for this. In any other case the problem was handled in hotspot